tests/visitor_test.py

```diff
@@ -0,0 +1,99 @@
+import re
+
+import pytest
+
+from dead import find_unused
+from dead import Visitor
+
+
+@pytest.fixture()
+def visitor() -> Visitor:
+    return Visitor()
+
+
+@pytest.mark.parametrize(
+    ('files', 'error'),
+    (
+        # unused variable
+        (
+            {'a.py': ('x_variable = True\n', False)},
+            r'x_variable is never read, defined in .*\.py:1',
+        ),
+
+        # unused function
+        (
+            {'a.py': ('\n\ndef x_function():\n    pass', False)},
+            r'x_function is never read, defined in .*\.py:3',
+        ),
+
+        # unused class
+        (
+            {'a.py': ('\nclass XClass():\n    pass', False)},
+            r'XClass is never read, defined in .*\.py:2',
+        ),
+
+        # used in same module
+        (
+            {'a.py': ('x_variable = True\nprint(x_variable)', False)},
+            False,
+        ),
+
+        # used in other module
+        (
+            {
+                'a.py': ('x_variable = True\n', False),
+                'b.py': ('import a\nprint(a.x_variable)', False),
+            },
+            False,
+        ),
+        (
+            {
+                'a.py': ('x_variable = True\n', False),
+                'b.py': ('from a import x_variable', False),
+            },
+            False,
+        ),
+
+        # used in tests
+        (
+            {
+                'a.py': ('x_variable = True\n', False),
+                'a_test.py': ('from a import x_variable', True),
+            },
+            r'x_variable is only referenced in tests, defined in .*\.py:1',
+        ),
+
+        # disable comment
+        (
+            {'a.py': ('\nclass XClass():  "# dead: disable"\npass', False)},
+            r'XClass is never read, defined in .*\.py:2',
+        ),
+        (
+            {'a.py': ('x_variable = True  # dead:disable', False)},
+            False,
+        ),
+        (
+            {'a.py': ('\nclass XClass():  # noqa dead: disable\n    pass', False)},
+            False,
+        ),
+
+        # disabled comment for multiline definition
+        (
+            {'a.py': ('\ndef x_function(  # noqa dead: disable\n):\n    pass', False)},
+            False,
+        ),
+    ),
+)
+def test_visitor_defines(tmpdir, visitor, files, error):
+    for filename, (src, is_test) in files.items():
+        path = tmpdir.join(filename)
+        path.write(src)
+        visitor.visit_file(path, is_test=is_test)
+
+    errors = list(find_unused(visitor))
+
+    if error:
+        assert errors
+        assert re.search(error, errors[0])
+    else:
+        assert not errors
```

Comment:
don't write logic in tests, this should be two separate tests

Reply:
done